nexus/src/external_api/http_entrypoints.rs

```diff
@@ -1010,6 +1024,256 @@ async fn organization_projects_put_project_policy(
+    let nexus = &apictx.nexus;
+    let pool_params = pool_params.into_inner();
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let pool = nexus.ip_pool_create(&opctx, &pool_params).await?;
+        Ok(HttpResponseCreated(IpPool::from(pool)))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Fetch a single IP Pool.
+#[endpoint {
+    method = GET,
+    path = "/ip-pools/{pool_name}",
+    tags = ["ip-pools"],
+}]
+async fn ip_pools_get_ip_pool(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<IpPoolPathParam>,
+) -> Result<HttpResponseOk<views::IpPool>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let pool_name = &path.pool_name;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let pool = nexus.ip_pool_fetch(&opctx, pool_name).await?;
+        Ok(HttpResponseOk(IpPool::from(pool)))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Delete an IP Pool.
+#[endpoint {
+    method = DELETE,
+    path = "/ip-pools/{pool_name}",
+    tags = ["ip-pools"],
+}]
+async fn ip_pools_delete_ip_pool(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<IpPoolPathParam>,
+) -> Result<HttpResponseDeleted, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let pool_name = &path.pool_name;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        nexus.ip_pool_delete(&opctx, pool_name).await?;
+        Ok(HttpResponseDeleted())
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Update an IP Pool.
+#[endpoint {
+    method = PUT,
+    path = "/ip-pools/{pool_name}",
+    tags = ["ip-pools"],
+}]
+async fn ip_pools_put_ip_pool(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<IpPoolPathParam>,
+    updates: TypedBody<params::IpPoolUpdate>,
+) -> Result<HttpResponseOk<views::IpPool>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let pool_name = &path.pool_name;
+    let updates = updates.into_inner();
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let pool = nexus.ip_pool_update(&opctx, pool_name, &updates).await?;
+        Ok(HttpResponseOk(pool.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Parameters for controlling pagination of IP Pool ranges
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, JsonSchema)]
+pub struct IpPoolRangeScanParams {
+    #[serde(default = "default_ip_pool_range_order")]
+    direction: PaginationOrder,
```

Comment:
I mentioned it because I believe this is the first appearance of `PaginationOrder` in the OpenAPI schema, which I think means it's the only API that lets you do this?  I don't mind having it but it's more code to maintain and test and we can't rip it out later without breaking the API. 🤷

Reply:
I did _not_ realize it's the only endpoint that supports it. I can remove, most of the cleanup here was to use the `paginated` function.